platform/src/main/java/org/hillview/utils/DateParsing.java

```diff
@@ -123,11 +124,11 @@ public DateParsing(String s) {
 
         this.parseAsDate = false;
         // none of the standard formats worked, let's try some custom ones
-        for (String regexp : DATE_FORMAT_REGEXPS.keySet()) {
-            if (s.toLowerCase().matches(regexp)) {
-                String format = DATE_FORMAT_REGEXPS.get(regexp);
+        for (Map.Entry<String, String> regexpEntry : DATE_FORMAT_REGEXPS.entrySet()) {
```

Comment:
This loop has around 40 iterations, so this is overkill.

Reply:
Doesn't hurt. :)